### PR TITLE
Test namespace validation

### PR DESCRIPTION
**Pull request by 2uasimojo**

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Note to reviewers: Again, this sits on top of the big pep8 change (PR #47), so best reviewed by looking at just commit f266433a3a0b062eaec2c89a28a8f43a7439fc6d.

**Comment by lisa** [comment]
I'm not sure the code changes for the linter's benefit are worth making the code harder to read for humans.

**Comment by lisa** [comment]
> Note to reviewers: Again, this sits on top of the big pep8 change (PR #47), so best reviewed by looking at just commit [f266433](https://github.com/openshift/managed-cluster-validating-webhooks/commit/f266433a3a0b062eaec2c89a28a8f43a7439fc6d).

Oops, does that mean https://github.com/openshift/managed-cluster-validating-webhooks/pull/51#issuecomment-611103714 is best fit into #47?

**Comment by 2uasimojo** [comment]
> > Note to reviewers: Again, this sits on top of the big pep8 change (PR #47), so best reviewed by looking at just commit [f266433](https://github.com/openshift/managed-cluster-validating-webhooks/commit/f266433a3a0b062eaec2c89a28a8f43a7439fc6d).
> 
> Oops, does that mean [#51 (comment)](https://github.com/openshift/managed-cluster-validating-webhooks/pull/51#issuecomment-611103714) is best fit into #47?

Probably. I'll answer there.

**Comment by jewzaam** [comment]
Because built on top of https://github.com/openshift/managed-cluster-validating-webhooks/pull/47 which isn't merged and needs updates:
/hold

**Comment by 2uasimojo** [comment]
> Because built on top of #47 which isn't merged and needs updates:
> /hold

Rebased and moved off of #47 to get it moving.

File: src/webhook/test_namespace_validation.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        # *and* the 'dedicated-admins' group.
+        for ns in PRIVILEGED_NAMESPACES:
+            for gl in GROUP_LISTS:
+                # Always include dedicated-admins
+                groups = gl + ('dedicated-admins',)
+                self.runtest(ns, groups, False)
+
+    def test_allow_group(self):
+        # If the group list doesn't contain 'dedicated-admins', always ALLOW,
+        # even if the namespace is privileged. (Really?)
```
Comment (jewzaam) [inline src/webhook/test_namespace_validation.py:99]: There's a problem here this brings up, tracked in https://issues.redhat.com/browse/OSD-3511
Beyond scope of this PR?

Comment (2uasimojo) [reply src/webhook/test_namespace_validation.py:99]: I definitely feel that's for a different PR, yes.

(For that matter, I'm pretty unconcerned about the unrealistic-ness of some of the sample data we're testing with above. This is unit testing the code without making any assumptions about how the back end _really_ works.)

**Pull request merged**